base_layer/core/src/transactions/transaction_protocol/sender.rs

```diff
@@ -490,23 +491,42 @@ impl SenderTransactionProtocol {
         sender_offset_private_key: &PrivateKey,
         output: &TransactionOutput,
         commitment_factory: &CommitmentFactory,
-    ) -> Result<ComSignature, TPE> {
+    ) -> Result<ComAndPubSignature, TPE> {
         // Create sender signature
-        let public_commitment_nonce = PublicKey::from_secret_key(private_commitment_nonce);
-        let e = output.get_metadata_signature_challenge(Some(&public_commitment_nonce));
-        let sender_signature = Signature::sign_raw(sender_offset_private_key, private_commitment_nonce.clone(), &e)?;
-        let sender_signature = sender_signature.get_signature();
+        let ephemeral_public_nonce = PublicKey::from_secret_key(private_commitment_nonce);
+        let sender_signature = TransactionOutput::create_sender_partial_metadata_signature(
+            output.version,
+            &output.commitment,
+            output.metadata_signature.ephemeral_commitment(),
+            &output.script,
+            &output.features,
+            sender_offset_private_key,
+            Some(private_commitment_nonce),
+            &output.covenant,
+            &output.encrypted_value,
+            output.minimum_value_promise,
+        )?;
         // Create aggregated metadata signature
-        let (r_pub, u, v) = output.metadata_signature.complete_signature_tuple();
-        let r_pub_aggregated = r_pub + &public_commitment_nonce;
-        let u_aggregated = u + sender_signature;
-        let aggregated_metadata_signature = ComSignature::new(r_pub_aggregated, u_aggregated, v.clone());
-
-        let sender_offset_public_key = PublicKey::from_secret_key(sender_offset_private_key);
+        let aggregated_metadata_signature = &sender_signature + &output.metadata_signature;
```

Comment:
It's possible that the recipient maliciously provided its own values for `ephemeral_pubkey` or `u_y` before returning it to the sender. Provided the output's sender offset public key cannot be overwritten by the recipient in a way that forces the sender to include it in the challenge, this will be caught by the verification check below, but could also be short-circuited by asserting that the recipient's partial metadata signature has `ephemeral_pubkey == u_y == 0`.

Reply:
It may also be a good idea to include this check anyway for defense in depth.

Reply:
When doing m-of-n this will need to be bypassed. I prefer a one fit for all check. 

But yeah if the recipient changes `ephemeral_pubkey` or `u_y` the check below will fail and the signature will not pass.

The recipient can at all times replace  `sender offset public key` and create a new signature at will, but this will break the `script_offset` of the transaction.